src/Util/PhpServicesCreator.php

```diff
@@ -0,0 +1,610 @@
+
+                return 'inline('.$className.')';
+            }
+            $args = \is_array($value->getValue())
+                ? $this->createListFromArray($value->getValue())
+                : $this->toString($value->getValue());
+
+            return $value->getTag().'('.$args.')';
+        }
+
+        switch (\gettype($value)) {
+            case 'array':
+                return $this->createArrayArgument($value);
+                break;
+            case 'string':
+                return $this->createStringArgument($value);
+                break;
+            case 'boolean':
+                // The convertor preserve the boolean values only if it's necessary. >>>
+                if ($preserveValueIfTrueBoolean && true === $value) {
+                    return 'true';
+                }
+                // >>> Because most of the methods don't need to pass true as an argument.
+                return true === $value ? '' : 'false';
+                break;
+            default:
+                return $value;
+        }
+    }
+
+    private function createArrayArgument(array $values): string
+    {
+        if (1 === \count($values)) {
+            return $this->toString($values[0]);
+        }
+
+        return $this->isAssociativeArray($values)
+            ? $this->createAssociativeArray($values)
+            : $this->createSequentialArray($values);
+    }
+
+    private function createSequentialArray(array $array, bool $transformInList = false): string
+    {
+        if ($this->indentationIsRequired($array)) {
+            $arguments = "[\n";
+            foreach ($array as $subKey => $subValue) {
+                $arguments .= $this->tab(3).$this->toString($subValue, true).",\n";
+            }
+
+            return $arguments.$this->tab(2).']';
+        }
+
+        $stringArray = true === $transformInList ? '' : '[';
+
+        foreach ($array as $key => $value) {
+            $stringArray .= $this->toString($value, true).', ';
+        }
+        $stringArray = rtrim($stringArray, ', ');
+
+        return $transformInList ? $stringArray : $stringArray.']';
+    }
+
+    private function createAssociativeArray(array $array): string
+    {
+        if ($this->indentationIsRequired($array)) {
+            $stringArray = "[\n";
+            foreach ($array as $key => $value) {
+                $stringArray .= $this->tab(4).$this->toString($key).' => '.$this->toString($value, true);
+                $stringArray .= \count($array) > 1 ? ",\n" : "\n";
+            }
+
+            return $stringArray.$this->tab(3).']';
+        }
+
+        $stringArray = '[';
+        foreach ($array as $key => $value) {
+            $stringArray .= $this->toString($key).' => '.$this->toString($value, true);
+            $stringArray .= next($array) ? ', ' : '';
+        }
+
+        return $stringArray.']';
+    }
+
+    private function createListFromArray(array $array): string
```

Comment:
I don't normally add phpdoc to private methods. But I think it would be helpful to add some basic phpdoc - maybe some examples - to the following methods:

* createListFromArray
* createStringArgument
* toString
* createSequentialArray
* createArrayArgument

These are commonly used, and they're complex enough that it's hard to follow the code, because what they do isn't totally clear.

Reply:
Done.